core/src/main/java/org/bitcoinj/uri/BitcoinURI.java

```diff
@@ -186,12 +185,13 @@ public BitcoinURI(@Nonnull Network network, String input) throws BitcoinURIParse
         }
 
         // Attempt to parse the rest of the URI parameters.
-        parseParameters(network, addressToken, nameValuePairTokens);
+        parseParameters(network, nameValuePairTokens);
 
+        String addressToken = addressSplitTokens[0];  // may be empty!
```

Comment:
I think moving this line is confusing and also unrelated to removing the unused parameter.  i.e. the code reads much better if we can easily see that the address is before the split (`[0]`) and the other tokens are after the split (`[1]`)